python/ray/serve/backend_state_manager.py

```diff
@@ -118,6 +118,11 @@ def start_or_update(self, backend_info: BackendInfo):
 
     def check_ready(self) -> bool:
         ready, _ = ray.wait([self._ready_obj_ref], timeout=0)
+        # In case of deployment constructor failure, ray.get will help to
+        # surface exception to each update() cycle.
+        if len(ready) > 0:
+            ray.get(ready)
```

Comment:
I'd prefer to return a boolean indiciating constructor failure rather than using the exception as the signaling mechanism here. That more closely matches the and would avoid errors where there is an exception in this code that trigger us to treat it as a constructor failure.

Reply:
That indeed matches the current exception as signaling mechanism that I saw from current codebase, and i would like to keep it consistent :) 

I wonder if there're better ways to throw initial exception to deploy() call's wait_for_goal() however, I assume ideally we want to be able to surface and set the original exception from constructor to the value of goal_id (which is an asyncio.Future now) ? But update() in backend_state handles it initially and we will lose it before reaching user code. Maybe we can have a mix of controller level global exception handling for cases like this that we want to set exception to the goal_id we're waiting for ?